design-proposals/klip-24-key-column-semantics-in-queries.md

```diff
@@ -541,28 +564,51 @@ The following naming strategies were rejected:
 
 #### `ROWKEY`/`JOINKEY` or other hardcoded name
 
-The synthesised column would take on a hardcoded name. This name would need to be reserved to avoid
-name clashes.
+The synthesised column would take on a hardcoded name. This was rejected as an approach due to
+complexities in the code required to avoid name clashing.  To explain this, consider the following
+worked example.
 
-`ROWKEY` was rejected as existing users will have sources with key columns called `ROWKEY`.
-Reserving this name would cause the statements building these sources to fail post-upgrade.synthesised
+First, let us remember that ksqlDB already supports including unambiguous source columns in a join's
+projection _without_ their source qualifier, for example:
 
-`JOINKEY` or some other hardcoded name was rejected primarliy due to the implementation issues
-mentioned above: not only would it be possible to validate the presence of `JOINKEY` in the
-projection, the user would also be free to use `JOINKEY` in other places, e.g. as a parameter to a
-UDF or within a binary expression, etc.  This would not work by default and would require extra
-effort to make work, including additional steps in the streams topology - there seems little ROI on
-such effort or the complexity it will add to the code.
+```sql
+-- I1 schema: ID KEY, V0, V1
+-- I2 schema: ID KEY, V0, V2
+CREATE TABLE OUTPUT AS
+   SELECT I1.ID, I2.V0, V1, V2 FROM I1 JOIN I2 ON I1.ID = I2.ID;
+-- OUTPUT schema: I1_ID KEY, I2_V0, V1, V2
+```
+
+In the above query, note how `I1.ID` and `I2.V0` must include their source to avoid being ambiguous,
+however `V1` and `V2` do not.
 
-#### `KSQL_COL_x`
+Now let us consider how this would play out if synthetic join keys had a hardcoded name `ROWKEY`:
 
-The synthesised column would take on a generated name in the form `KSQL_COL_x`, where `x` is an
-integer and the name is guaranteed to be unique, i.e. to not clash with any other system generated
-column names.
+```sql
+CREATE TABLE TABLE_WITH_ROWKEY AS
+    SELECT * FROM I1 FULL OUTER JOIN I2 ON I1.ID = I2.ID;
+-- T1: ROWKEY KEY, I1_ID, I1_V0, I1_V1, I2_ID, I2_V0, I2_V1
+
+CREATE TABLE OUTPUT AS
+    SELECT ROWKEY, [...] FROM TABLE_WITH_ROWKEY I3 FULL OUTER JOIN I4 ON I3.ROWKEY = I4.ID;
+```
 
-This was rejected as the name would be hard for the user to know by looking at the query, i.e. the
-presence of other columns with generated names may affect the name of the synthesised column. This
-was deemed to make this solution unworkable.
+The first statement executes without issue and results in a table with a schema that includes the
+hardcoded `ROWKEY` column name. However, the presence of `ROWKEY` second statement's projection
+posses a problem: does this `ROWKEY` pertain to the synthetic join key this query introduces or
+`TABLE_WITH_ROWKEY.ROWKEY`?
```

Comment:
The only way what it could be is, that it revers to the generated one. One can prefix the `TABLE_WITH_ROKWEY.ROWKEY` column but the newly generated one cannot be prefix. -- But I agree that it might not be clear (even if not ambiguous).